### ISSUE TITLE
cargo fix cmd no longer works

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#5358]
The [https://github.com/rust-lang/rust-clippy#automatically-applying-clippy-suggestions](readme) states we can run `cargo fix -Z unstable-options --clippy` but this no longer works:
```
error: Found argument '--clippy' which wasn't expected, or isn't valid in this context

USAGE:
    cargo fix -Z <FLAG>...

For more information try --help
```

Maybe it's related to https://github.com/rust-lang/cargo/pull/7533 ?
EDIT: looks like it: https://github.com/rust-lang/cargo/pull/7533#issuecomment-599133278

**Comment by flip1995** [comment]
ping @yaahc Do you have an idea, what's going on here and what we can do about this?

**Comment by yaahc** [comment]
My guess is that cargo fix was shelling out to clippy preview which got removed and now it needs to use RUSTC_WORKSPACE_WRAPPER instead now, I can open a PR to fix it, guess this means my PR finally merged into nightly 😅

**Comment by flip1995** [comment]
Thanks! It would be great, if you could provide a fix for this.

**Comment by yaahc** [comment]
Okay so I just checked and it looks like cargo-fix isn't an external subcommand anymore, its built into cargo, and I think I remember we decided that we didn't want to have mentions of clippy in cargo itself which is what this PR does https://github.com/rust-lang/cargo/pull/7533, it completely removes all special case integration for clippy and adds the functionality to cargo that clippy would need to be implemented as an external subcommand.

So the plan going forward is to have fix be invoked by clippy's external subcommand `cargo clippy --fix` and not support `cargo fix --clippy` for the time being. If we wanted to have a clippy option on fix it would have to go back to knowing how to detect if `clippy-driver` is available and would have to pick a version which is I think something we're aiming to avoid.

I'm currently working on the `cargo clippy --fix` version, sorry for the inconvenience, this particular bit of integration has gone through a few iterations between october and now as we figured out the best way to structure everything.

so yea short term @matthiaskrgr you should be able to still apply them on nightly by running

```shell
RUSTC_WORKSPACE_WRAPPER=/path/to/clippy-driver cargo fix -Zunsable-options
```

Which is essentially how the new cargo-clippy subcommand will work going forward.

**Comment by yaahc** [comment]
Here we go, WIP

https://github.com/rust-lang/rust-clippy/pull/5363

**Comment by ebroto** [comment]
Closing this as it was fixed by #5363.

I will add the missing help for the `--fix` command after/if rust-lang/cargo#8976 is merged in a stabilization PR here in Clippy :crossed_fingers:
